spec/unit/hanami/cli/commands/app/generate/action_spec.rb

```diff
@@ -1345,6 +1345,37 @@ def handle(request, response)
         end
       end
     end
+
+    context "when run from within a slice directory" do
+      let(:slice) { "books" }
+      let(:controller) { "users" }
+      let(:action) { "index" }
+      let(:action_name) { "#{controller}.#{action}" }
+
+      it "creates the action for the current slice" do
+        within_application_directory do
+          prepare_slice!
+
+          # That is a lot of mocks, is there a better way for this?
+          mock_app = double("Hanami::App")
+          allow(mock_app).to receive(:root).and_return(Pathname.new(Dir.pwd))
+          allow(mock_app).to receive(:slices).and_return({slice.to_sym => double("Slice")})
+          allow(mock_app).to receive(:namespace).and_return(app)
+          allow(mock_app).to receive(:inflector).and_return(inflector)
+
+          allow(subject).to receive(:app).and_return(mock_app)
+
+          slice_path = File.join(Dir.pwd, "slices", slice)
+          allow(Dir).to receive(:pwd).and_return(slice_path)
+
+          subject.call(name: action_name)
+
+          expect(fs.exist?("slices/#{slice}/actions/#{controller}/#{action}.rb")).to be(true)
```

Comment:
Those checks are happening a lot in the codebase. Wondering if a verbose, custom rspec matcher would do some good on those?
```ruby
expect(an_action_to_exist(action, slice: slice))
```
and such
or even, with some custom chain methods
```ruby
expect(an_action_to_exist(action).in_slice(slice))
```

I know not everyone is a fan of custom rspec matches, but I quite like them, coming from a place where I think RSpec is a big, heavy tool, that by default is just a base spec framework to me modified in according to project needs. Leveraging those custom matchers etc. is to me one of the main reasons to use RSpec and writing specs instead of tests using some smaller test framework.

Reply:
Hmm that's an interesting idea! I say leave it for now, but let's keep it in mind as we build this out.